### PR TITLE
[fix](crash) be crash because of duplicate __PARTIAL_UPDATE_AUTO_INC_COLUMN__

### LINKED ISSUE #41171: [Bug] Be crash because of __PARTIAL_UPDATE_AUTO_INC_COLUMN__
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

2.1.x

### What's Wrong?

Be crash:
```
(gdb) bt
#0  doris::vectorized::IColumn::append_data_by_selector_impl<doris::vectorized::ColumnVector<long> > (this=0x7f43d8bbc410, res=..., selector=..., begin=0, 
    end=164) at /root/be/src/vec/columns/column_impl.h:45
#1  0x000055d3597f7a20 in doris::vectorized::Block::append_to_block_by_selector (this=0x7f43d54c1af0, dst=0x7f4613874000, selector=...)
    at /root/be/src/vec/core/block.cpp:822
#2  0x000055d35fd4f13f in doris::vectorized::VNodeChannel::add_block (this=0x7f43d59fb590, block=0x7f43d54c1af0, payload=payload@entry=0x7f43e110a390)
    at /root/be/src/vec/sink/writer/vtablet_writer.cpp:520
#3  0x000055d35fd5908e in doris::vectorized::VTabletWriter::write (this=this@entry=0x7f43dc3da000, input_block=...)
    at /root/be/src/vec/sink/writer/vtablet_writer.cpp:1720
#4  0x000055d35fd58aaf in doris::vectorized::VTabletWriter::_send_new_partition_batch (this=this@entry=0x7f43dc3da000)
    at /root/be/src/vec/sink/writer/vtablet_writer.cpp:1364
#5  0x000055d35fd597be in doris::vectorized::VTabletWriter::_do_try_close (this=this@entry=0x7f43dc3da000, state=<optimized out>, exec_status=...)
    at /root/be/src/vec/sink/writer/vtablet_writer.cpp:1381
#6  0x000055d35fd5a2fc in doris::vectorized::VTabletWriter::close (this=0x7f43dc3da000, exec_status=...)
    at /root/be/src/vec/sink/writer/vtablet_writer.cpp:1489
#7  0x000055d35fcd77b3 in _ZN5doris10vectorized15AsyncWriterSinkINS0_13VTabletWriterEXadsoKcL_ZNS0_16VOLAP_TABLE_SINKEEEEE5closeEPNS_12RuntimeStateENS_6StatusE (this=this@entry=0x7f4616237560, state=state@entry=0x7f4613485400, exec_status=...) at /root/be/src/vec/sink/async_writer_sink.h:120
#8  0x000055d35fcd6838 in doris::vectorized::VOlapTableSink::close (this=0x7f4616237560, state=0x7f4613485400, exec_status=...)
    at /root/be/src/vec/sink/volap_table_sink.cpp:131
#9  0x000055d356934ad5 in doris::PlanFragmentExecutor::open_vectorized_internal (this=this@entry=0x7f4616211090)
    at /root/be/src/runtime/plan_fragment_executor.cpp:355
#10 0x000055d356934036 in doris::PlanFragmentExecutor::open (this=this@entry=0x7f4616211090) at /root/be/src/runtime/plan_fragment_executor.cpp:273
#11 0x000055d356935918 in doris::PlanFragmentExecutor::execute (this=0x7f4616211090) at /root/be/src/runtime/plan_fragment_executor.cpp:403
#12 0x000055d356881b14 in doris::FragmentMgr::_exec_actual(std::shared_ptr<doris::PlanFragmentExecutor>, std::function<void (doris::RuntimeState*, doris::Status*)> const&) (this=this@entry=0x7f46cdec4800, fragment_executor=..., cb=...) at /root/be/src/runtime/fragment_mgr.cpp:478
#13 0x000055d356891b7e in doris::FragmentMgr::exec_plan_fragment(doris::TExecPlanFragmentParams const&, std::function<void (doris::RuntimeState*, doris::Status*)> const&)::$_0::operator()() const (this=<optimized out>) at /root/be/src/runtime/fragment_mgr.cpp:781
#14 std::__invoke_impl<void, doris::FragmentMgr::exec_plan_fragment(doris::TExecPlanFragmentParams const&, std::function<void (doris::RuntimeState*, doris::Status*)> const&)::$_0&>(std::__invoke_other, doris::FragmentMgr::exec_plan_fragment(doris::TExecPlanFragmentParams const&, std::function<void (doris::RuntimeState*, doris::Status*)> const&)::$_0&) (__f=...) at /var/local/ldb-toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61
#15 std::__invoke_r<void, doris::FragmentMgr::exec_plan_fragment(doris::TExecPlanFragmentParams const&, std::function<void (doris::RuntimeState*, doris::Status*)> const&)::$_0&>(doris::FragmentMgr::exec_plan_fragment(doris::TExecPlanFragmentParams const&, std::function<void (doris::RuntimeState*, doris::Status*)> const&)::$_0&) (__fn=...) at /var/local/ldb-toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:111
#16 std::_Function_handler<void (), doris::FragmentMgr::exec_plan_fragment(doris::TExecPlanFragmentParams const&, std::function<void (doris::RuntimeState*, doris::Status*)> const&)::$_0>::_M_invoke(std::_Any_data const&) (__functor=...)
    at /var/local/ldb-toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:291
#17 0x000055d356b44728 in doris::ThreadPool::dispatch_thread (this=0x7f46f25a4800) at /root/be/src/util/threadpool.cpp:543
#18 0x000055d356b39ae1 in std::function<void ()>::operator()() const (this=0x424b0053cd)
    at /var/local/ldb-toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:560
#19 doris::Thread::supervise_thread (arg=0x7f46ce3d36e0) at /root/be/src/util/thread.cpp:498
#20 0x00007f46f2976ea5 in ?? ()
#21 0x0000000000000000 in ?? ()
```

the direct reason is in: doris::vectorized::Block::append_to_block_by_selector
```
(gdb) f 1
#1  0x000055d3597f7a20 in doris::vectorized::Block::append_to_block_by_selector (this=0x7f43d54c1af0, dst=0x7f4613874000, selector=...)
    at /root/be/src/vec/core/block.cpp:822
822     in /root/be/src/vec/core/block.cpp

(gdb) pvector this->data 22 (end with 2  __PARTIAL_UPDATE_AUTO_INC_COLUMN__ columns)
Vector size = 24

(gdb) pvector dst->_names 22 (end with 1  __PARTIAL_UPDATE_AUTO_INC_COLUMN__ columns)
Vector size = 23
```

### What You Expected?

do not crash

### How to Reproduce?

Cannot reproduce, maybe related to:

- AUTO PARTITION
- "store_row_column" = "true"
- Partial Update

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by cambyzju**
Issue Number: close #41171

come from #26915

### CODE REVIEW

**Review by lide-reed — DISMISSED** [review]
LGTM

**Review by liutang123 — DISMISSED** [review]
LGTM

**Comment by cambyzju** [comment]
run buildall

**Comment by cambyzju** [comment]
run buildall

**Review by lide-reed — DISMISSED** [review]
LGTM

File: be/src/vec/sink/vtablet_block_convertor.cpp [hunk 1]
```diff
@@ -568,10 +569,15 @@ Status OlapTableBlockConvertor::_partial_update_fill_auto_inc_cols(vectorized::B
     for (size_t i = 0; i < rows; i++) {
         dst_values.emplace_back(_auto_inc_id_allocator.next_id());
     }
+    // in VRowDistribution we reuse _batching_block, so the input block may already include PARTIAL_UPDATE_AUTO_INC_COL,
```
Comment (zhannngchen) [inline be/src/vec/sink/vtablet_block_convertor.cpp:572]: When you got here, which means the `block` object is from `_batching_block`, it already have __PARTIAL_UPDATE_AUTO_INC_COL__, so we don't need to call the method  `_partial_update_fill_auto_inc_cols()`
we'd better to change the logic at L80

Comment (cambyzju) [reply be/src/vec/sink/vtablet_block_convertor.cpp]: done

**Comment by cambyzju** [comment]
run buildall

**Review by zclllyybb — APPROVED** [review]
LGTM

**Review by zhannngchen — APPROVED** [review]
LGTM

**Pull request merged**